### DISCUSSION TITLE
Help me out with this Python Devs

### DISCUSSION THREAD

**Discussion by ghost** [community/community#88418] [Programming Help]
### Body

So actually I have a repository named "_Ideas-For-The-Weekend-Python-Edition_", [link for it](https://github.com/Yuheshpandian/Ideas-For-The-Weekend-Python-Edition). That contains interesting python project ideas to do in the weekend as the name suggests. I want some 🐍 python devs to help me to come out with new project ideas for it...as it is really you can even contribute/collaborate, or even star it this would be actually very helpful to me in learning about collaborations/contibutions. If you could help me pls 🥹 do.

Thank you to all in advance 🙏🏻

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sasha-korovkina** [comment]
Hey there! I'm Sasha, I work as a python developer in the finance sector. Here are some things which we wish we had time to develop, so I'll share them with you :) 

**Python x Text**
1. The problem: we want to name a file according to its contents. Pythonic solution: you can develop something to read the file (PDF file would be pretty easy) and use NLP to name the file. 
2. Mapping: there are a lot of times in finance when we need to map stuff. For example - a stock can be called 'Apple' on one exchange and 'Apple Inc' on the other. These are the same. Pythonic solution: make a Python algorithm that recognizes these and calls them the same name.
3. Address recognition: so often, we have a file which contains a bunch of text and addresses in it. Unfortunately, most libraries can't recognize the addresses explicitly. I'd love for there to be a tool which does this.
4. Make logging more fun: I try to throw in some jokes into our logging or when we have certain weird actions within our programs. Having a Python script (maybe linked to some LLM) which generates this jokes would be great. On the topic of this - build a joke generator!

Let me know if these helped! I have plenty of other ideas so hmu if you wanna chat, also I'd really appreciate a follow :)

**Reply by ghost** [reply]
Yeah these would be awesome for the intermediate or advanced sections for the repo. You really helpful. If you would like to help I will post the discussions link so you could help me with some ideas...

**Comment by xserts** [comment]
Great repository you earned a star! Try adding something like a simple quiz game.

**Reply by ghost** [reply]
Can you be a little more specific

**Reply by xserts** [reply]
A game quiz game that asks a question and lets choose one of the 4 answers. You can add a part to it that lets users add their own questions. It should be doable in one weekend.

**Reply by ghost** [reply]
Ok that was really helpful

**Reply by ghost** [reply]
I will surely add This

**Reply by ghost** [reply]
Added the project you suggested, you can now see it in the [Repository](https://github.com/Yuheshpandian/Ideas-For-The-Weekend-Python-Edition/tree/main/Ideas/Quiz_Game). 

Thank you for the Idea

**Comment by anekobtw** [comment]
Hey! Tbh, a very good idea, I like it. You may add more games such as Sudoku or Tetris. Additionally, including projects related to APIs of other apps (like YouTube, Spotify, Instagram, etc.) would be awesome. Good luck in improving your repository!

**Reply by ghost** [reply]
Thank you so much, I will add those...

**Comment by ghost** [comment]
Just a note for all, your ideas are my content. But its not easy to create all by My self. So if you have an idea and you can or you has create(d) it. Then try to add those by contributing. This will save much time and we could create awesome projects. Till that brainstorm ideas...

Thank you
